packages/babel-plugin-transform-optional-chaining/test/fixtures/execute/new.js

```diff
@@ -0,0 +1,65 @@
+let calls = 0;
+const obj = {
+  a: {
+    b: class {
+      constructor(val) {
+        assert.equal(val, 1);
+        assert(this instanceof obj.a.b);
+        calls++;
+      }
+    },
+  },
+  c: class {
+    constructor(val) {
+      assert.equal(val, 1);
+      assert(this instanceof obj.c);
+      calls++;
+    }
+  },
+};
+
+let ab = new obj?.a?.b(1);
+assert(ab instanceof obj.a.b);
```

Comment:
Hm... if this was an optional path, this could result in `instanceof void 0`, which is a runtime error. Maybe something to support in the spec, or just leave it as an error? cc/ @gisenberg